src/diff/index.js

```diff
@@ -314,13 +315,7 @@ function diffElementNodes(
 			// if newVNode matches an element in excessDomChildren or the `dom`
 			// argument matches an element in excessDomChildren, remove it from
 			// excessDomChildren so it isn't later removed in diffChildren
-			if (
-				child != null &&
-				((newVNode.type === null
-					? child.nodeType === 3
-					: child.localName === newVNode.type) ||
-					dom == child)
-			) {
+			if (child != null && (dom == child || child.localName == nodeType)) {
```

Comment:
I'm leaving this as a suggestion to avoid the comment push re-running benchmarks:

```suggestion
			//
			// Note: This takes advantage of Text nodes having `.localName=undefined`,
			// which is loosely equal to Text VNodes' `.type=null`. Elements use string equality.
			if (child != null && (dom == child || child.localName == nodeType)) {
```